packages/angular/ssr/node/src/app-engine.ts

```diff
@@ -41,28 +73,25 @@ export interface AngularNodeServerAppManager {
  * and optionally transforms index HTML before rendering.
  */
 export class AngularNodeAppEngine extends AngularAppEngine implements AngularNodeServerAppManager {
-  /**
-   * Renders an HTTP response based on the incoming request using the Angular server application.
-   *
-   * The method processes the incoming request, determines the appropriate route, and prepares the
-   * rendering context to generate a response. If the request URL corresponds to a static file (excluding `/index.html`),
-   * the method returns `null`.
-   *
-   * Example: A request to `https://www.example.com/page/index.html` will render the Angular route
-   * associated with `https://www.example.com/page`.
-   *
-   * @param request - The incoming HTTP request object to be rendered. It can be a `Request` or `IncomingMessage` object.
-   * @param requestContext - Optional additional context for the request, such as metadata or custom settings.
-   * @returns A promise that resolves to a `Response` object, or `null` if the request URL is for a static file
-   * (e.g., `./logo.png`) rather than an application route.
-   */
   override render(
     request: Request | IncomingMessage,
     requestContext?: unknown,
   ): Promise<Response | null> {
-    const webReq = request instanceof Request ? request : createWebRequestFromNodeRequest(request);
+    return super.render(this.toWebRequest(request), requestContext);
+  }
+
+  override getHeaders(
+    request: Request | IncomingMessage,
+  ): Readonly<Record<string, string>> | undefined {
```

Comment:
Consider: Can we drop the `| undefined` case here and just return `{}` in this case?